drivers/sensor/tmp116/tmp116.h

```diff
@@ -24,13 +25,19 @@
 #define TMP116_DEVICE_ID		0x1116
 #define TMP117_DEVICE_ID		0x0117
 
+#define DEFINE_TMP116(_num) \
```

Comment:
This should go in your .c file, it should be used to initialize compile-time const items in your `device->config`.

There are a number of examples in the codebase to help clarify. E.g. 
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/i2c/i2c_emul.c#L137

For someone coming from the Linux world, the link above might shed some light on things as it highlights Zephyr's usage of DeviceTree to bake in compile-time const values into ROM and how devices are compile-time const pointers (If a device is not referenced, its section can be discarded).

The documentation could probably be improved, but here is the link to that.
https://docs.zephyrproject.org/latest/reference/drivers/index.html#c.DEVICE_DT_INST_DEFINE

Reply:
I have done all the changes, now I see that both device->config and device->data will be generated using the device tree by the preprocessor for each instance.